src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/xy.ts

```diff
@@ -0,0 +1,661 @@
+  { meta: { description: 'Reference line layer for ES|QL queries' } }
+);
+
+/**
+ * Common properties for all annotation types
+ */
+const annotationEventShared = {
+  name: schema.maybe(schema.string({ meta: { description: 'Display name for the annotation' } })),
+  color: schema.maybe(staticColorSchema),
+  hidden: schema.maybe(schema.boolean({ meta: { description: 'Whether to hide the annotation' } })),
+};
+
+/**
+ * Visual properties specific to point annotations
+ */
+const annotationPointShared = {
+  ...annotationEventShared,
+  icon: schema.maybe(
+    schema.string({ meta: { description: 'Icon to display at the annotation point' } })
+  ),
+  line: schema.maybe(
+    schema.object(
+      {
+        stroke_width: schema.number({ meta: { description: 'Vertical line width in pixels' } }),
+        stroke_dash: schema.oneOf(
+          [schema.literal('straight'), schema.literal('dashed'), schema.literal('dotted')],
+          { meta: { description: 'Vertical line style' } }
+        ),
+      },
+      { meta: { description: 'Vertical line configuration for point annotation' } }
+    )
+  ),
+};
+
+/**
+ * Text label configuration for annotations
+ */
+const annotationTextLabelOption = schema.object(
+  {
+    type: schema.literal('label'),
+    text: schema.string({ meta: { description: 'Label text to display' } }),
+  },
+  { meta: { description: 'Text label for annotation' } }
+);
+
+/**
+ * Timestamp format for annotations (Unix timestamp or ISO date string)
+ */
+const annotationTimestampSchema = schema.oneOf([
+  schema.number({ meta: { description: 'Unix timestamp in milliseconds' } }),
+  schema.string({ meta: { description: 'ISO date string' } }),
+]);
+
+/**
+ * Query-based annotation that finds events matching a filter
+ */
+const annotationQuery = schema.object(
+  {
+    ...annotationPointShared,
+    type: schema.literal('query'),
+    query: filterSchema,
+    time_field: schema.string({ meta: { description: 'Field containing the timestamp' } }),
+    text: schema.maybe(annotationTextLabelOption),
+    extra_fields: schema.maybe(
+      schema.arrayOf(
+        schema.string({ meta: { description: 'Additional field to include in tooltip' } }),
+        { meta: { description: 'Additional fields for annotation tooltip' } }
+      )
+    ),
+  },
+  { meta: { description: 'Annotation from query results matching a filter' } }
+);
+
+/**
+ * Manually specified point annotation at a specific timestamp
+ */
+const annotationManualEvent = schema.object(
+  {
+    ...annotationPointShared,
+    type: schema.literal('point'),
+    timestamp: annotationTimestampSchema,
+    text: schema.maybe(
+      schema.oneOf([
+        annotationTextLabelOption,
+        schema.object(
+          {
+            type: schema.literal('field'),
+            field: schema.string({ meta: { description: 'Field name containing label text' } }),
+          },
+          { meta: { description: 'Text from document field' } }
+        ),
+      ])
+    ),
```

Comment:
Can we revisit this?
I rather prefer to ahve two fields, one called: `label: string` and the other `useFieldAslabel: field` then having something like this where I need to describe `text: {type: 'field', 'field':'message'}` mostly because I see it cumbersome having to specify a type to describe a text label

Reply:
Fixed